azurelinuxagent/ga/exthandlers.py

```diff
@@ -486,33 +491,44 @@ def handle_enable(self, ext_handler_i):
             ext_handler_i.update_settings()
 
         ext_handler_i.enable()
+        # Remove the UNINSTALL_FAILED from env variables list after calling Install and Enable
+        ext_handler_i.remove_key_from_handler_env(UNINSTALL_FAILED)
 
     @staticmethod
     def _update_extension_handler(old_ext_handler_i, ext_handler_i):
 
-        def execute_old_handler_command(func):
+        def execute_old_handler_command(op, func):
             """
             Created a common wrapper to execute all commands that need to be executed from the old handler
             so that it can have a common exception handling mechanism
+            :param op: The current running operation
             :param func: The command to be executed on the old handler
             """
+            continue_on_update_failure = False
             try:
+                continue_on_update_failure = ext_handler_i.load_manifest().is_continue_on_update_failure()
                 func()
             except ExtensionError as e:
                 # Reporting the event with the old handler and raising a new ExtensionUpdateError to set the
                 # handler status on the new version
-                msg = ustr(e)
+                msg = "%s; ContinueOnUpdate: %s" % (ustr(e), continue_on_update_failure)
                 old_ext_handler_i.report_event(message=msg, is_success=False)
-                raise ExtensionUpdateError(msg)
+                if not continue_on_update_failure:
+                    raise ExtensionUpdateError(msg)
+
+                logger.info("Continue on Update failure flag is set, proceeding with update")
+                error_env_name = DISABLE_FAILED if op == WALAEventOperation.Disable else UNINSTALL_FAILED
+                ext_handler_i.set_handler_env_variable({error_env_name: 'True'})
 
-        execute_old_handler_command(lambda: old_ext_handler_i.disable())
+        execute_old_handler_command(op=WALAEventOperation.Disable, func=lambda: old_ext_handler_i.disable())
         ext_handler_i.copy_status_files(old_ext_handler_i)
         if ext_handler_i.version_gt(old_ext_handler_i):
             ext_handler_i.update()
+            # Remove the DISABLE_FAILED from env variables list after calling Update
+            ext_handler_i.remove_key_from_handler_env(DISABLE_FAILED)
```

Comment:
I need this line still because we documented to the extension publishers that the DISABLE_FAILED environment variable would only be available for update command. If the update command was successful, we need to remove it from the handler_env to ensure that we dont send it to any other commands (install/enable/uninstall)

Reply:
You need to protect this against exceptions in the code above this; for example, if update() raises you wouldn't remove the variable

Reply:
Done